src/adminDashboard/adminDashboard.ts

```diff
@@ -51,14 +57,100 @@ export class AdminDashboard {
   async hydrate(): Promise<void> {
     if (this.ethereumService.defaultAccountAddress) {
       const defaultAccount = this.ethereumService.defaultAccountAddress.toLowerCase();
-      this.seeds = this.seedService.seedsArray
-        .filter((seed) => { return seed.admin.toLowerCase() === defaultAccount;});
+      this.seeds = this.seedService.seedsArray;
     } else {
       this.seeds = [];
     }
   }
 
-  connect():void {
+  async setSeed(index) {
+    this.selectedSeed = this.seeds[index];
+  }
+
+  async fundSeed() {
+    try {
+      await this.selectedSeed.projectTokenContract.transfer(this.selectedSeed.address, this.selectedSeed.seedAmountRequired);
+    } catch (error) {
+      alert(error.message);
+    }
+  }
+
+  async retrieveProjectTokens(receiver: Address) {
+    try {
+      await this.selectedSeed.contract.callStatic.retrieveSeedTokens(receiver);
+      await this.selectedSeed.contract.retrieveSeedTokens(receiver);
+    } catch (error) {
+      alert(error.message);
+    }
+  }
+
+  async withdrawFundingTokens() {
+    try {
+      await this.selectedSeed.contract.callStatic.withdraw();
+      await this.selectedSeed.contract.withdraw();
+    } catch (error) {
+      alert(error.message);
+    }
+  }
+
+  async addWhitelist() {
+    try {
+      const filterPattern = /([^\W]+)/g;
+      const response = await axios.get(this.selectedSeed.metadata.seedDetails.whitelist);
```

Comment:
please use our WhitelistService for fetching, adding a list, and adding and removing addresses.

Reply:
done